scripts/importmetadata

```diff
@@ -0,0 +1,229 @@
+# Checks for all output .go resources with a declared Importer: in schema
+# Checks if it uses standard tpgresource.ParseImportId() functions, if so converts that regex
+# Checks if uses schema.ImportStatePassThrough(), if it has a hardcoded return based on how ID is constructed in the resource
+# Anything else is non standard -- will take from MD files 
+#
+# Expecting to see a resource and it's not being generated? Please reach out to @nickelliot
+
+shopt -s nullglob
+
+if [ -z "$1" ]
+then
+	echo "Please run script with the format './scripts/importmetadata <ga or beta> <basepath (optional arg)>'"
+	exit 1
+fi
+if [ -z "$2" ]
+then
+	basepath="$GOPATH/src/github.com/hashicorp/"
+else
+	basepath=$2
+fi
+if [ "$1"=="beta" ]
+then
+	basepath="${basepath}/terraform-provider-google-beta"
+	servicepath="${basepath}/google-beta"
+else
+	basepath="${basepath}/terraform-provider-google"
+	servicepath="${basepath}/google"
+fi
+
+echo "# $resource_type: [list-of-acceptable-import-templates]" > importmetadata.txt
+echo "{" >> importmetadata.txt
+
+for file in $(find $servicepath/services/**/resource_*.go -type f -print); do
+	importable=$(ggrep -Pzo 'Importer:' $file)
+	product=
+	shortfile="$(basename $file .go)"
+	resource=${shortfile/resource/google}
+
+#does not work for GCP meta resources and we need to prune the extra google
+	resource=${resource/google_google_/google_}
+
+    if [ -z "$importable"  ]
+    then
+
+##############################################
+#resources with shared importing configuration
+#logging has imports configured in MD files
+		importable=$(ggrep -Pzo 'return ResourceLoggingBucketConfig' $file)
+		if [ -z "$importable"  ]
+    	then
+#			echo -n \	\"$resource\"\: >> importmetadata.txt
+#			echo \	not importable >> importmetadata.txt
+
+# NOT IMPORTABLE RESOURCES
+			continue
+		else
+#logging bucket config uses MD files
+#needs to extract formats from MD files
+			resourcenogoog=${resource/google_/}
+			resourcemd=$basepath/website/docs/r/$resourcenogoog.html.markdown
+	    	if [ -f "$resourcemd" ]
+	    	then
+				importformat=$(ggrep -Pzo '```\n\$ terraform import(.*\n)*?```' $resourcemd)
+				if [ -z "$importformat"  ]
+	    		then
+					echo -n \	\"$resource\"\: >> importmetadata.txt
+					echo \	NON STANDARD MD LOGGING >> importmetadata.txt
+#	    		else
+					echo -n \	\"$resource\"\: >> importmetadata.txt
+					importformat=$(sed -e 's|``` \$ terraform import [^.]*\.[^[[:space:]]* |\[\"|g'  <<< $importformat)
+					importformat=$(sed -e 's| \$ terraform import [^.]*\.[^[[:space:]]* |", "|g'  <<< $importformat)
+					importformat=$(sed -e 's| ```|\"\]|g'  <<< $importformat)
+			    	echo \ "$importformat" >> importmetadata.txt
+		    	fi
+		    fi
+		fi
+##############################################
+    else
+
+# checks for tpgresource.parseimportid, standard output for all .yaml generated resources
+		importformat=$(ggrep -Pzo 'tpgresource\.ParseImportId\(\n*?[ \t]*?\[]string{\n*?([ \t]*".*?",\n)*.*?}' $file)
```

Comment:
Imports sourced through this method include the start/end of input markers, we should strip those for consistency with other formats. For example:

```
"google_compute_forwarding_rule": ["^projects/{{project}}/regions/{{region}}/forwardingRules/{{name}}$", "^{{project}}/{{region}}/{{name}}$", "^{{region}}/{{name}}$", "^{{name}}$"]
```